### PR TITLE
Fix zookeeper and nacos issues working as registry, metadata and configcenter

### PR DESCRIPTION
**Pull request by chickenlj**
As titled, I tested dubbo-go Nacos and Zookeeper implementation working as registry, metadata, and configcenter in dubbo admin. 

The changes in this pull request are bugfixes made to different parts that finally make admin work.

### CODE REVIEW

File: registry/servicediscovery/service_mapping_change_listener_impl.go [hunk 1]
```diff
@@ -63,6 +65,9 @@ func (lstn *ServiceMappingChangedListenerImpl) OnEvent(e observer.Event) error {
 		err error
 		reg registry.Registry
 	)
+
+	lstn.mux.Lock()
```
Comment (Lvnszn) [inline registry/servicediscovery/service_mapping_change_listener_impl.go:69]: 这个加锁之后中间 return 没有 unlock

Comment (chickenlj) [reply registry/servicediscovery/service_mapping_change_listener_impl.go:69]: 我来加上defer

File: metadata/report/delegate/delegate_report.go [hunk 2]
```diff
@@ -19,6 +19,7 @@ package delegate
 
 import (
 	"encoding/json"
+	gxset "github.com/dubbogo/gost/container/set"
```
Comment (Lvnszn) [inline metadata/report/delegate/delegate_report.go:22]: go get -u github.com/dubbogo/tools/cmd/imports-formatter
imports-formatter

File: Makefile [hunk 3]
```diff
@@ -68,6 +68,11 @@ license: clean prepareLic
 .PHONY: verify
 verify: clean license test
 
+.PHONE: fmt
+fmt:
+	$(GO_GET) -u github.com/dubbogo/tools/cmd/imports-formatter
+	imports-formatter
+
```
Comment (justxuewei) [inline Makefile:75]: Please add this function to a new pull request. Keep a pull request to implement only one function as much as possible to make code review easier.

**Comment by justxuewei** [comment]
Could you split this pull request into several pull requests? Or could you give some description about this pull request?

File: Makefile (hunk 3 above)
Comment (chickenlj) [reply Makefile:75]: I will create a new pull request for this part.

**Comment by chickenlj** [comment]
> 

Please see the description I leave above.  

I understand it would be ideal if we submit a pull request containing only changes fixing exactly one issue. In my case here, I don't have a very clear boundary of which issue I am targeting, so the standard I use is to make Nacos being able to work together with Admin.

**Review by zhaoyunxing92 — APPROVED** [review]

**Pull request merged**